[WIP] Adding a JOSS paper

choldgraf opened the pull request:
**[Here is a link to the latest paper draft](https://hackmd.io/BwZgJmDGBsCcYFoCGAzEBGBAWWKBGySArJgAxhJ4CmY0II0qQA==)**
This adds the `paper.md` document required by JOSS for submission.

There are still a couple of points to finish up before it's ready to go:

- [ ] Release v0.1 of BinderHub
- [ ] Confirm author list (I just used the list in `team-compass` + the grant)
- [ ] Include ORCID information for all authors
- [ ] Flesh out and confirm summary
- [ ] Announce this to the listserv per the Jupyter submission docs

More information is here: https://github.com/jupyterhub/binderhub/issues/421

### ORCID checklist:

- [x] Tim Head
- [ ] Jessica Forde
- [x] Brian Granger
- [x] Chris Holdgraf
- [x] M Pacer
- [x] Yuvi Panda
- [x] Fernando Perez
- [x] Min Regan Kelley
- [x] Carol Willing

choldgraf commented:
@betatim @jzf2101 @yuvipanda @minrk and @willingc could you all either paste here or update the PR with your ORCIDs?

choldgraf commented:
ORCID IDs for @ellisonbg and @fperez would be helpful as well.

Once we get all these ORCIDs in there, I can make a PR somewhere else so that we have that info in a persistent place

ellisonbg commented:
Thanks!

My ORCID is: 0000-0002-5223-6168

choldgraf commented:
Note: see the checklist in https://github.com/jupyterhub/binderhub/pull/428#issue-292594409 for a list of who we still need ORCID numbers for

willingc commented on paper.md:
```diff
@@ -46,18 +46,29 @@ bibliography: paper.bib
+repository, then serve that Docker image to users via a public link. BinderHub
+will flexibly create cloud resources as users request to interact with a
+particular repository's link, and will automatically destroy these resources
+after periods of inactivity. A public deployment of BinderHub exists at
```
Is it "a period of inactivity"?

ellisonbg commented on paper.md:
```diff
@@ -0,0 +1,74 @@
+particular repository's link, and will automatically destroy these resources
+after periods of inactivity. A public deployment of BinderHub exists at
+`mybinder.org`, though BinderHub is designed to be deployed by anyone. It is
+cloud-agnostic and can support many different workflows.
```
Even though binder is a very generic way of hosting/deploying a wide range of things, I think it will help us to be a bit more explicit about the main focus - of enabling Jupyter notebooks to be run by anyone anywhere. We should also mention other things (RStudio) etc. as well. 

We should also connect Binder to the broader goals of reproducibility.

ellisonbg replied:
I am more than willing to take an editing pass at some point if that would help.

yuvipanda replied:
@ellisonbg I think we've explicitly tried to make sure (and message!) that Binder is not Jupyter specific, so it'd be good to keep that messaging here in some form.

choldgraf replied:
incoming push tries to clarify specific use-cases a bit more, LMKWYT when it lands

choldgraf commented:
+1 please feel free to commit to the pr!

On Wed, Jan 31, 2018, 1:12 PM Brian E. Granger <notifications@github.com>
wrote:

> *@ellisonbg* commented on this pull request.
> ------------------------------
>
> In paper.md
> <https://github.com/jupyterhub/binderhub/pull/428#discussion_r165188108>:
>
> > +---
> +
> +# Summary
> +
> +BinderHub enables authors to create sharable, interactive
> +computational environments by specifying their environment in an online
> +git repository.
> +
> +BinderHub runs a service that lets users put their code in a repository online, use
> +the BinderHub service to build a Docker image using configuration files in that
> +repository, then serve that Docker image to users via a public link. BinderHub
> +will flexibly create cloud resources as users request to interact with a
> +particular repository's link, and will automatically destroy these resources
> +after periods of inactivity. A public deployment of BinderHub exists at
> +`mybinder.org`, though BinderHub is designed to be deployed by anyone. It is
> +cloud-agnostic and can support many different workflows.
>
> I am more than willing to take an editing pass at some point if that would
> help.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/jupyterhub/binderhub/pull/428#discussion_r165188108>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABwSHTUd2enYDMDamWB-n0wB_mZ8DdHqks5tQNdNgaJpZM4RxgOR>
> .
>

yuvipanda commented:
https://orcid.org/0000-0002-8055-2870 is mine!

mpacer commented:
My ORCID is [0000-0002-6680-2941](http://orcid.org/0000-0002-6680-2941).

betatim commented:
I am https://orcid.org/0000-0003-0931-3698

betatim commented on paper.md:
```diff
@@ -0,0 +1,74 @@
+particular repository's link, and will automatically destroy these resources
+after periods of inactivity. A public deployment of BinderHub exists at
+`mybinder.org`, though BinderHub is designed to be deployed by anyone. It is
+cloud-agnostic and can support many different workflows.
```
"many different workflows" -> I would either add one or two concrete examples or remove this. What kind of "workflow" do we mean with this? RStudio and notebooks and ...?

As is it feels like a fluffy catch-all//caveat emptor that even the authors didn't really know what it covers ;)

choldgraf replied:
see the latest push for a clarification on this, LMKWYT!

betatim commented on paper.md:
```diff
@@ -0,0 +1,74 @@
+ - name: Project Jupyter
+   orcid:
+   affiliation:
+ - name: Tim Head
```
🔡 Head comes after Forde and Granger

choldgraf replied:
lol I can haz english?

choldgraf commented on paper.md:
```diff
@@ -0,0 +1,74 @@
+
+BinderHub interfaces closely with the [JupyterHub Helm Chart](https://github.com/jupyterhub/zero-to-jupyterhub-k8s),
+which runs a JupyterHub service on Kubernetes. It includes a Python module
+that interfaces with other open-source tools (e.g., [repo2docker](https://github.com/jupyter/repo2docker)) to run the service
```
tried improving this language a bit, LMK

choldgraf commented:
ok, some small comments pushed per @willingc , @ellisonbg , and @betatim ...thanks for the input! 

Whole team: Please LMK what you think, mention if you'd like to see more edits, or take an editing pass of your own! (just ping this PR before you start editing so others know not to work on it during that time)

pinging @jzf2101 and @fperez : what's your ORCID?

choldgraf commented:
Two notes:

1. Unless somebody protests, I'm going to delete all the comments that only list their ORCID number - we've got that in the PR now and I don't want the issue to get too cluttered.
2. If somebody would like to take a pass at the language, that'd be useful. We can re-use some of it for the scipy 2018 submission so having it nailed down sooner than later would be helpful

ellisonbg commented:
Hi all, thanks to everyone working on this. I wanted to clarify my comments
about the "many different workflows" language. I do think it is important
for us to describe binder in a way that is inclusive to different usage
cases other than Jupyter notebooks. However the original language was a bit
to abstract to the point that I don't think the reader would understand the
connection between "many different workflows" and the actual reality of how
binder is used in practice. One way for us to be more specific would be to
add language about the focus being on "interactive computing workflows".
While binder could be used to deploy non-interactive computing systems, I
would say that is the one thing that ties the main practical applications
together. Then we could give examples of the types of interactive computing
systems binder works with so far and list Jupyter, RStudio, nteract, etc..
Thoughts?



On Tue, Feb 6, 2018 at 11:29 AM, Chris Holdgraf <notifications@github.com>
wrote:

> Two notes:
>
>    1. Unless somebody protests, I'm going to delete all the comments that
>    only list their ORCID number - we've got that in the PR now and I don't
>    want the issue to get too cluttered.
>    2. If somebody would like to take a pass at the language, that'd be
>    useful. We can re-use some of it for the scipy 2018 submission so having it
>    nailed down sooner than later would be helpful
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/jupyterhub/binderhub/pull/428#issuecomment-363537455>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AABr0FMvSJzFQHT1KITrPRRDSIIdjmt3ks5tSKgXgaJpZM4RxgOR>
> .
>



-- 
Brian E. Granger
Associate Professor of Physics and Data Science
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com

choldgraf commented:
hey @ellisonbg - I edited the paragraph a couple days back to try and be more specific about use-cases. take a look and LMKWYT

https://github.com/jupyterhub/binderhub/pull/428/files#diff-fb6416b20e925ad59e9f77c28b957d88R58

ellisonbg commented on paper.md:
```diff
@@ -0,0 +1,78 @@
+particular repository's link, and will automatically destroy these resources
+after a period of inactivity. While BinderHub is designed to fit a number of
+cloud-based workflows, it has a particular focus on improving reproducibility
+for scientific work, enabling interactive published artifacts of results, and
```
thanks, this is an improvement, but I think it is still a bit abstract. Thinking out loud...

"it has a particular focus on improving the reproducibility of published artifacts that involve interactive scientific computations, such as those done with Jupyter Notebooks, RMarkdown, Shiny, Jupyter Widgets, etc." 

"Some usage cases that binder enables:

* Open-source projects can use binder to allow anyone in the world to try their software interactively, with zero installation
* Educators can deploy computational lessons...
* Research groups can...
"

choldgraf replied:
I like the idea of a bulleted-list...I'll give that a shot in the next push!

choldgraf commented:
I just realized that a much more efficient way to iterate on this text is to edit collaboratively on a hackmd prior to including it in this PR. **I have removed `paper.md` from this PR, and suggest we all keep iterating at the hackmd at [this link](https://hackmd.io/BwZgJmDGBsCcYFoCGAzEBGBAWWKBGySArJgAxhJ4CmY0II0qQA==)**. I've also updated the top-level comment to reflect this.

choldgraf commented:
@ellisonbg I also took a spin at including your comments about use-cases in the hackmd (see above comment for context)

ellisonbg commented:
Sounds good, I will have a look after midterms tomorrow!

On Wed, Feb 7, 2018 at 12:12 PM, Chris Holdgraf <notifications@github.com>
wrote:

> @ellisonbg <https://github.com/ellisonbg> I also took a spin at including
> your comments about use-cases in the hackmd (see above comment for context)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/jupyterhub/binderhub/pull/428#issuecomment-363895643>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AABr0Pe8IEF5Hli3GkQaG-SeiB-xBlRBks5tSgO-gaJpZM4RxgOR>
> .
>



-- 
Brian E. Granger
Associate Professor of Physics and Data Science
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com

ellisonbg commented:
Does hackmd allow comments?

On Wed, Feb 7, 2018 at 2:59 PM, Brian Granger <ellisonbg@gmail.com> wrote:

> Sounds good, I will have a look after midterms tomorrow!
>
> On Wed, Feb 7, 2018 at 12:12 PM, Chris Holdgraf <notifications@github.com>
> wrote:
>
>> @ellisonbg <https://github.com/ellisonbg> I also took a spin at
>> including your comments about use-cases in the hackmd (see above comment
>> for context)
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/jupyterhub/binderhub/pull/428#issuecomment-363895643>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/AABr0Pe8IEF5Hli3GkQaG-SeiB-xBlRBks5tSgO-gaJpZM4RxgOR>
>> .
>>
>
>
>
> --
> Brian E. Granger
> Associate Professor of Physics and Data Science
> Cal Poly State University, San Luis Obispo
> @ellisonbg on Twitter and GitHub
> bgranger@calpoly.edu and ellisonbg@gmail.com
>



-- 
Brian E. Granger
Associate Professor of Physics and Data Science
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com

choldgraf commented:
not that I know of, but feel free to just put `**[BG]: <comment>** or something like that...whatever works for you.

(for longer papers or artifacts I agree this could become clunky...not sure what's the best worfklow in that case)

jzf2101 commented:
I don't see the paper?

choldgraf commented:
@jzf2101 see https://github.com/jupyterhub/binderhub/pull/428#issuecomment-363894236

choldgraf commented:
Hi all - just a note that v0.1 of BinderHub is now live: https://github.com/jupyterhub/binderhub/releases/tag/0.1.0

we originally planned on making a JOSS paper when this happened. However maybe we want to re-visit since v0.1 is a purely "maintenance release" that we don't want to advertise much.

Two questions:

1. Do we still want to publish a JOSS paper?
2. Do we want to wait until v0.2?

jzf2101 commented:
I think we should at least wait until v0.2?

betatim commented:
Closing this as this hasn't received any attention for about a year. If interest is rekindled we can re-open this or start a new PR.

choldgraf commented:
yeah I agree, I have been leaving this open in the hope that we might one day do it, but I don't have the bandwidth to push it forward so I'm +1 on closing :-/
